packages/core/src/fhir-to-cda/cda-templates/components/immunizations.ts

```diff
@@ -0,0 +1,200 @@
+import { Bundle, Immunization, Location } from "@medplum/fhirtypes";
+import { findResourceInBundle, isImmunization, isLocation } from "../../../external/fhir/shared";
+import { CdaSimpleReference, ObservationTableRow } from "../../cda-types/shared-types";
+import {
+  buildAddressText,
+  buildCodeCe,
+  buildCodeCvFromCodeableConcept,
+  buildInstanceIdentifier,
+  buildPerformerFromLocation,
+  formatDateToCdaTimestamp,
+  formatDateToHumanReadableFormat,
+  getDisplaysFromCodeableConcepts,
+  withoutNullFlavorObject,
+} from "../commons";
+import {
+  NOT_SPECIFIED,
+  extensionValue2015,
+  hl7actCode,
+  loincCodeSystem,
+  loincSystemName,
+  oids,
+  placeholderOrgOid,
+} from "../constants";
+import { createTableRowsAndEntries } from "../create-table-rows-and-entries";
+import { initiateSectionTable } from "../table";
+import { AugmentedImmunization } from "./augmented-resources";
+
+const immunizationsSectionName = "immunizations";
+
+const tableHeaders = ["Immunization", "Status", "Location", "Date"];
+
+export function buildImmunizations(fhirBundle: Bundle) {
+  const immunizations: Immunization[] =
+    fhirBundle.entry?.flatMap(entry => (isImmunization(entry.resource) ? [entry.resource] : [])) ||
+    [];
+
+  if (immunizations.length === 0) {
+    return undefined;
+  }
+
+  const augmentedImmunizations = createAugmentedImmunizations(immunizations, fhirBundle);
+
+  const { trs, entries } = createTableRowsAndEntries(
+    augmentedImmunizations,
+    createTableRowFromImmunization,
+    createEntryFromEncounter
+  );
+
+  const table = initiateSectionTable(immunizationsSectionName, tableHeaders, trs);
+
+  return {
+    templateId: buildInstanceIdentifier({
+      root: oids.immunizationsSection,
+    }),
+    code: buildCodeCe({
+      code: "11369-6",
+      codeSystem: loincCodeSystem,
+      codeSystemName: loincSystemName,
+      displayName: "History of immunizations",
+    }),
+    title: "IMMUNIZATIONS",
+    text: table,
+    entry: entries,
+  };
+}
+
+function createAugmentedImmunizations(
```

Comment:
just OOC why augmenting immunization with location? Like there are a bunch of different resources referenced in immunization. Why specifically did we choose location?

Reply:
I looked at a bunch of CDAs and what I'm mapping for the Immunization section is similar to those. Location was fairly common, so I included it as well.